src/coreclr/jit/gentree.cpp

```diff
@@ -3483,7 +3483,10 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
                 break;
 
             case GT_CNS_DBL:
-                level = 0;
+            {
+                var_types targetType = tree->TypeGet();
+                level                = 0;
+#if defined(TARGET_XARCH)
                 /* We use fldz and fld1 to load 0.0 and 1.0, but all other  */
```

Comment:
This comment is off for x86/x64. We use `xorps`/`xorpd` for zero. There is no special variant for `1.0`, only "all bits set", but we don't use it outside SIMD due to it being a non-default NaN and only really useful for masking.

Reply:
I apparently already commented this last time I reviewd: https://github.com/dotnet/runtime/pull/44419#discussion_r520206743
😄